src/shredder/reed_solomon.rs

```diff
@@ -30,20 +30,30 @@ pub(super) fn reed_solomon_shred(
     num_coding: usize,
 ) -> Result<(Vec<DataShred>, Vec<CodingShred>), ReedSolomonShredError> {
     let Slice {
+        parent,
         slot,
         slice_index,
         is_last,
         merkle_root: _,
         data,
     } = slice;
-    reed_solomon_shred_raw(*slot, *slice_index, *is_last, data, num_data, num_coding)
+    reed_solomon_shred_raw(
+        *parent,
+        *slot,
+        *slice_index,
+        *is_last,
+        data,
+        num_data,
+        num_coding,
+    )
 }
 
 /// Splits the given data into `num_data` data shreds, then generates
 /// `num_coding` additional Reed-Solomon coding shreds.
-/// The slice-specific fields `slot`, `slice_index`, and `is_last` are included
-/// in every `DataShred` or `CodingShred`.
+/// The slice-specific fields `parent`, `slot`, `slice_index`, and `is_last`
+/// are included in every `DataShred` or `CodingShred`.
 pub(super) fn reed_solomon_shred_raw(
```

Comment:
@qkniep: if I introduce the field in the naive way like I did here, then we will be sending the parent info in each shred in the slice.  So that will be 40 bytes extra per shred I suppose.  Is that problematic you think and something we should optimise?